Modules/_sre/sre.c

```diff
@@ -533,6 +622,13 @@ state_fini(SRE_STATE* state)
     /* See above PyMem_Free() for why we explicitly cast here. */
     PyMem_Free((void*) state->mark);
     state->mark = NULL;
+    /* SRE_REPEAT pool */
+    repeat_pool_clear(state);
+#ifdef Py_DEBUG
+    if (pattern) {
+        pattern->fail_after_count = -1;
+    }
+#endif
```

Comment:
How about remove `pattern->fail_after_count = -1;` here?
This counter is only for debuging purpose, maybe removing the resetting is irrelevant.

Then there won't be these changes everywhere in this patch:
```diff
-       state_fini(&state);
+       state_fini(&state, self);
```